modules/aardvarkBidAdapter.js

```diff
@@ -0,0 +1,154 @@
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'aardvark';
+const DEFAULT_ENDPOINT = 'bidder.rtk.io';
+const SYNC_ENDPOINT = 'sync.rtk.io';
+const AARDVARK_TTL = 300;
+const AARDVARK_CURRENCY = 'USD';
+
+var hasSynced = false;
+
+export function resetUserSync() {
+  hasSynced = false;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+
+  isBidRequestValid: function(bid) {
+    return ((typeof bid.params.ai === 'string') && !!bid.params.ai.length &&
+        (typeof bid.params.sc === 'string') && !!bid.params.sc.length);
+  },
+
+  buildRequests: function(validBidRequests, bidderRequest) {
+    var auctionCodes = [];
+    var requests = [];
+    var requestsMap = {};
+    var referer = utils.getTopWindowUrl();
+    var pageCategories = [];
+
+    if (window.rtkcategories && Array.isArray(window.rtkcategories)) {
+      pageCategories = window.rtkcategories;
+    }
+
+    utils._each(validBidRequests, function(b) {
+      var rMap = requestsMap[b.params.ai];
+      if (!rMap) {
+        rMap = {
+          shortCodes: [],
+          payload: {
+            version: 1,
+            jsonp: false,
+            rtkreferer: referer
+          },
+          endpoint: DEFAULT_ENDPOINT
+        };
+
+        if (pageCategories && pageCategories.length) {
+          rMap.payload.categories = pageCategories.slice(0);
+        }
+
+        if (b.params.categories && b.params.categories.length) {
+          rMap.payload.categories = rMap.payload.categories || []
+          utils._each(b.params.categories, function(cat) {
+            rMap.payload.categories.push(cat);
+          });
+        }
+
+        if (bidderRequest && bidderRequest.gdprConsent) {
+          rMap.payload.gdpr = bidderRequest.gdprConsent.gdprApplies;
```

Comment:
There may be rare chances where this value may be `undefined` if the CMP didn't have it defined.  Do you want to set your own back-up/default value in that case?  If so, I would recommend to check if `typeof bidderRequest.gdprConsent.gdprApplies === 'boolean'` so that you can accept natural cases where `gdprApplies` was set to `false.